### ISSUE TITLE
[BUG]Merlin XGBoost model training failure

### ISSUE THREAD

**Issue by zhiruiwang** [NVIDIA-Merlin/models#689]
### Bug description

When running XGBoost model example provided in [07-Train-traditional-ML-models-using-the-Merlin-Models-API](https://github.com/NVIDIA-Merlin/models/blob/main/examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb) notebook, I encounter an error regarding dask and DMatrix(See full error message in the Additional Context section).
All the other notebooks(01-06) with models based on Tensorflow, Implicit and lightfm execute successfully.

### Steps/Code to reproduce bug

In example notebook [07-Train-traditional-ML-models-using-the-Merlin-Models-API](https://github.com/NVIDIA-Merlin/models/blob/main/examples/07-Train-traditional-ML-models-using-the-Merlin-Models-API.ipynb), excute the code snippet by snippet until this chunk:
```
with Distributed():
    model = XGBoost(schema=schema_without_title, **xgb_booster_params)
    model.fit(
        train,
        evals=[(valid, 'validation_set'),],
        **xgb_train_params
    )
    metrics = model.evaluate(valid)
```

### Expected behavior

Model should train and return validation loss

### Environment details
- Merlin version: merlin-models==0.6.0 and the main branch
- Platform(Both didn't work): 
    - Azure Databricks Single-node Standard_NC6s_V3 GPU instance with Docker image: databricksruntime/gpu-rapids:cuda11
    - Google Colab with [RAPDIS.AI](https://colab.research.google.com/drive/1rY7Ln6rEE1pOlfSHCYOVaqt8OvDO35J0)
- Python version: 
    - Azure Databricks: 3.9.10
    - Google Colab: 3.7.13
- XGBoost version: 1.5.2

### Additional context

Full error message
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
File <command-3538978324726920>:3, in <module>
      1 with Distributed():
      2     model = XGBoost(schema=schema_without_title, **xgb_booster_params)
----> 3     model.fit(
      4         train,
      5         evals=[(valid, 'validation_set'),],
      6         **xgb_train_params
      7     )
      8     metrics = model.evaluate(valid)

File /databricks/python/lib/python3.9/site-packages/merlin/models/xgb/__init__.py:155, in XGBoost.fit(self, train, evals, use_quantile, **train_kwargs)
    149 if self.params.get("tree_method") == "gpu_hist" and use_quantile:
    150     # `DaskDeviceQuantileDMatrix` is a data type specialized
    151     # for the `gpu_hist` tree method that reduces memory overhead.
    152     # When training on GPU pipeline, it's preferred over `DaskDMatrix`.
    153     dmatrix_cls = xgb.dask.DaskDeviceQuantileDMatrix
--> 155 dtrain = dmatrix_cls(self.dask_client, X, label=y, qid=qid)
    156 watchlist = []
    158 if evals is None:

File /databricks/python/lib/python3.9/site-packages/xgboost/core.py:506, in _deprecate_positional_args.<locals>.inner_f(*args, **kwargs)
    504 for k, arg in zip(sig.parameters, args):
    505     kwargs[k] = arg
--> 506 return f(**kwargs)

File /databricks/python/lib/python3.9/site-packages/xgboost/dask.py:678, in DaskDeviceQuantileDMatrix.__init__(self, client, data, label, weight, base_margin, missing, silent, feature_names, feature_types, max_bin, group, qid, label_lower_bound, label_upper_bound, feature_weights, enable_categorical)
    657 @_deprecate_positional_args
    658 def __init__(
    659     self,
   (...)
    676     enable_categorical: bool = False,
    677 ) -> None:
--> 678     super().__init__(
    679         client=client,
    680         data=data,
    681         label=label,
    682         weight=weight,
    683         base_margin=base_margin,
    684         group=group,
    685         qid=qid,
    686         label_lower_bound=label_lower_bound,
    687         label_upper_bound=label_upper_bound,
    688         missing=missing,
    689         silent=silent,
    690         feature_weights=feature_weights,
    691         feature_names=feature_names,
    692         feature_types=feature_types,
    693         enable_categorical=enable_categorical,
    694     )
    695     self.max_bin = max_bin
    696     self.is_quantile = True

File /databricks/python/lib/python3.9/site-packages/xgboost/core.py:506, in _deprecate_positional_args.<locals>.inner_f(*args, **kwargs)
    504 for k, arg in zip(sig.parameters, args):
    505     kwargs[k] = arg
--> 506 return f(**kwargs)

File /databricks/python/lib/python3.9/site-packages/xgboost/dask.py:331, in DaskDMatrix.__init__(self, client, data, label, weight, base_margin, missing, silent, feature_names, feature_types, group, qid, label_lower_bound, label_upper_bound, feature_weights, enable_categorical)
    328 self.worker_map: Dict[str, "distributed.Future"] = defaultdict(list)
    329 self.is_quantile: bool = False
--> 331 self._init = client.sync(
    332     self._map_local_data,
    333     client,
    334     data,
    335     label=label,
    336     weights=weight,
    337     base_margin=base_margin,
    338     qid=qid,
    339     feature_weights=feature_weights,
    340     label_lower_bound=label_lower_bound,
    341     label_upper_bound=label_upper_bound,
    342 )

File /databricks/python/lib/python3.9/site-packages/distributed/utils.py:310, in SyncMethodMixin.sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    308     return future
    309 else:
--> 310     return sync(
    311         self.loop, func, *args, callback_timeout=callback_timeout, **kwargs
    312     )

File /databricks/python/lib/python3.9/site-packages/distributed/utils.py:364, in sync(loop, func, callback_timeout, *args, **kwargs)
    362 if error[0]:
    363     typ, exc, tb = error[0]
--> 364     raise exc.with_traceback(tb)
    365 else:
    366     return result[0]

File /databricks/python/lib/python3.9/site-packages/distributed/utils.py:349, in sync.<locals>.f()
    347     if callback_timeout is not None:
    348         future = asyncio.wait_for(future, callback_timeout)
--> 349     result[0] = yield future
    350 except Exception:
    351     error[0] = sys.exc_info()

File /databricks/python/lib/python3.9/site-packages/tornado/gen.py:762, in Runner.run(self)
    759 exc_info = None
    761 try:
--> 762     value = future.result()
    763 except Exception:
    764     exc_info = sys.exc_info()

File /databricks/python/lib/python3.9/site-packages/xgboost/dask.py:438, in DaskDMatrix._map_local_data(self, client, data, label, weights, base_margin, qid, feature_weights, label_lower_bound, label_upper_bound)
    435 await distributed.wait(parts)  # async wait for parts to be computed
    437 for part in parts:
--> 438     assert part.status == 'finished', part.status
    440 # Preserving the partition order for prediction.
    441 self.partition_order = {}

AssertionError: error
```

**Comment by radekosmulski** [comment]
Hi!

This error is coming from dask internals, hard to say if it is a compatibility issue with the version of dask you have or a bug.

The example runs okay in our up to date container against the main branch of Merlin Models. The container I ran this in to verify was `nvcr.io/nvidia/merlin/merlin-tensorflow:22.07`.

Not sure if this can be of help, but here are the versions of dask available in the container:
![image](https://user-images.githubusercontent.com/2444926/187336749-00b4b8a8-290e-458c-a29d-fb160b5bba21.png)

Closing as the issue is not due to the code in Merlin Models.
